From c0c6e4c1166c4868afc36649b9ed98081a6966e1 Mon Sep 17 00:00:00 2001
From: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>
Date: Fri, 24 Dec 2021 14:22:52 +0000
Subject: [PATCH 24/27] Comment mm_communicate failure log

When a getVariable() call is made with data size set to 0,
mm_communicate should return EFI_BUFFER_TOO_SMALL. This is
an expected behavior. There should not be any failure logs
in this case. So the error log is commented here.

Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 lib/efi_loader/efi_variable_tee.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_variable_tee.c b/lib/efi_loader/efi_variable_tee.c
index 67743d1f8fce..a34989efac83 100644
--- a/lib/efi_loader/efi_variable_tee.c
+++ b/lib/efi_loader/efi_variable_tee.c
@@ -411,7 +411,10 @@ static efi_status_t __efi_runtime mm_communicate(u8 *comm_buf, efi_uintn_t dsize
 	ret = ffa_mm_communicate(comm_buf, dsize);
 	#endif
 	if (ret != EFI_SUCCESS) {
-		log_err("%s failed!\n", __func__);
+		/* mm_communicate failure is logged even when getVariable() is called
+		 * with data size set to 0. This is not expected so logging is commented.
+		*/
+		//log_err("%s failed!\n", __func__);
 		return ret;
 	}
 
-- 
2.30.2

